src/wallet/rpcdump.cpp

```diff
@@ -723,6 +748,30 @@ UniValue dumpwallet(const UniValue& params, bool fHelp)
             file << strprintf(" # addr=%s\n", strAddr);
         }
     }
+
+    // Begin dump Zerocoins
+    list <CZerocoinEntry> listZerocoinEntries;
+    CWalletDB walletdb(pwalletMain->strWalletFile);
+    walletdb.ListPubCoin(listZerocoinEntries);
+
+    for (auto& zerocoinEntry : listZerocoinEntries) {
+        file << "zerocoin=1 ";
+        file << strprintf("%s ", zerocoinEntry.value.GetHex()); // value
+        file << strprintf("%s ", zerocoinEntry.denomination); // denomination
+        file << strprintf("%s ", zerocoinEntry.randomness.GetHex()); // randomness
+        file << strprintf("%s ", zerocoinEntry.serialNumber.GetHex()); // serialNumber
+        file << strprintf("%d ", zerocoinEntry.IsUsed); // IsUsed
+        file << strprintf("%s ", zerocoinEntry.nHeight); // nHeight
+        file << strprintf("%s ", zerocoinEntry.id); // id
+        if(!zerocoinEntry.ecdsaSecretKey.empty()){
+            uint256 ecdsaSecretKey;
+            std::copy(&zerocoinEntry.ecdsaSecretKey[0], &zerocoinEntry.ecdsaSecretKey[32], ecdsaSecretKey.begin());
+            file << strprintf("%s ", ecdsaSecretKey.GetHex()); // ecdsaSecretKey
```

Comment:
There is `HexStr` function so we don't need to use this way.

Reply:
Can you clarify what you mean?